src/app/modules/item/pages/item-edit/item-edit.component.ts

```diff
@@ -115,29 +115,33 @@ export class ItemEditComponent implements OnDestroy, PendingChangesComponent {
     );
   }
 
-  private getItemChanges(): ItemChanges {
+  private getItemChanges(): ItemChanges | undefined {
     const urlFormControl = this.itemForm.get('url');
     const usesApiFormControl = this.itemForm.get('uses_api');
     const textIdFormControl = this.itemForm.get('text_id');
 
-    if (urlFormControl === null || usesApiFormControl === null || textIdFormControl === null) return {};
+    if (urlFormControl === null || usesApiFormControl === null || textIdFormControl === null) return undefined;
 
-    const itemFormValues = {
-      url: urlFormControl.value as string | null,
-      uses_api: usesApiFormControl.value as boolean,
-      text_id: textIdFormControl.value as string,
-    };
+    const itemFormValues: ItemChanges = {};
 
-    if (itemFormValues.url === '') itemFormValues.url = null;
+    const url = urlFormControl.value as string;
+    if (url !== this.initialFormData?.url) itemFormValues.url = url !== '' ? url : null;
```

Comment:
As `url` can be `null` initially, if you do not change anything, it will always be different from the empty string... which means that even if there is no changes in the item itself (for instance, we only change the item strings), a request will be sent to the backend to update the url.

Reply:
fixed